tests/vmi_headless_test.go

```diff
@@ -163,6 +171,94 @@ var _ = Describe("[rfe_id:609][sig-compute]VMIheadless", func() {
 				Expect(vncCount).To(Equal(1), "should have exactly one VNC device")
 			})
 
+			It("should close connections after a while", func() {
+				getHandlerConnectionCount := func() int {
+					cmd := []string{"bash", "-c", fmt.Sprintf("ss -ntlap | grep %d | wc -l", virt_api.DefaultConsoleServerPort)}
+					stdout, stderr, err := tests.ExecuteCommandOnNodeThroughVirtHandler(virtClient, vmi.Status.NodeName, cmd)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(stderr).To(BeEmpty())
+
+					stdout = strings.TrimSpace(stdout)
+					stdout = strings.ReplaceAll(stdout, "\n", "")
+
+					handlerCons, err := strconv.Atoi(stdout)
+					Expect(err).ToNot(HaveOccurred())
+
+					return handlerCons
+				}
+
+				getRandomClientCall := func(vmi *v1.VirtualMachineInstance) func() {
```

Comment:
In general, adding randomness to e2e tests makes it difficult to reproduce and debug failures later on if some issues arise. Why not making this a predefined list of calls instead? Any benefit in having this randomized?

Reply:
Yeah on second thought that's very right.
Done